features/include/pcl/features/impl/gasd.hpp

```diff
@@ -0,0 +1,404 @@
+        hbin -= h0;
+
+        float v_xyzh1111 = v_xyz111 * hbin;
+        float v_xyzh1110 = v_xyz111 - v_xyzh1111;
+        float v_xyzh1101 = v_xyz110 * hbin;
+        float v_xyzh1100 = v_xyz110 - v_xyzh1101;
+        float v_xyzh1011 = v_xyz101 * hbin;
+        float v_xyzh1010 = v_xyz101 - v_xyzh1011;
+        float v_xyzh1001 = v_xyz100 * hbin;
+        float v_xyzh1000 = v_xyz100 - v_xyzh1001;
+        float v_xyzh0111 = v_xyz011 * hbin;
+        float v_xyzh0110 = v_xyz011 - v_xyzh0111;
+        float v_xyzh0101 = v_xyz010 * hbin;
+        float v_xyzh0100 = v_xyz010 - v_xyzh0101;
+        float v_xyzh0011 = v_xyz001 * hbin;
+        float v_xyzh0010 = v_xyz001 - v_xyzh0011;
+        float v_xyzh0001 = v_xyz000 * hbin;
+        float v_xyzh0000 = v_xyz000 - v_xyzh0001;
+
+        hists[grid_idx][h_idx] += v_xyzh0000;
+        hists[grid_idx][h_idx + 1] += v_xyzh0001;
+        hists[grid_idx + 1][h_idx] += v_xyzh0010;
+        hists[grid_idx + 1][h_idx + 1] += v_xyzh0011;
+        hists[grid_idx + (grid_size + 2)][h_idx] += v_xyzh0100;
+        hists[grid_idx + (grid_size + 2)][h_idx + 1] += v_xyzh0101;
+        hists[grid_idx + (grid_size + 3)][h_idx] += v_xyzh0110;
+        hists[grid_idx + (grid_size + 3)][h_idx + 1] += v_xyzh0111;
+        hists[grid_idx + (grid_size + 2) * (grid_size + 2)][h_idx] += v_xyzh1000;
+        hists[grid_idx + (grid_size + 2) * (grid_size + 2)][h_idx + 1] += v_xyzh1001;
+        hists[grid_idx + (grid_size + 2) * (grid_size + 2) + 1][h_idx] += v_xyzh1010;
+        hists[grid_idx + (grid_size + 2) * (grid_size + 2) + 1][h_idx + 1] += v_xyzh1011;
+        hists[grid_idx + (grid_size + 3) * (grid_size + 2)][h_idx] += v_xyzh1100;
+        hists[grid_idx + (grid_size + 3) * (grid_size + 2)][h_idx + 1] += v_xyzh1101;
+        hists[grid_idx + (grid_size + 3) * (grid_size + 2) + 1][h_idx] += v_xyzh1110;
+        hists[grid_idx + (grid_size + 3) * (grid_size + 2) + 1][h_idx + 1] += v_xyzh1111;
+
+        break;
+      }
+    }
+  }
+
+  //////////////////////////////////////////////////////////////////////////////////////////////
+  template <typename PointInT, typename PointOutT> void
+  GASDEstimation<PointInT, PointOutT>::copyHistogramsToOutput (int grid_size,
+                                                               int hists_size,
+                                                               vector<Eigen::VectorXf> &hists,
+                                                               PointCloudOut &output,
+                                                               int &pos)
+  {
+    for (int i = 0; i < grid_size; ++i)
+    {
+      for (int j = 0; j < grid_size; ++j)
+      {
+        for (int k = 0; k < grid_size; ++k)
+        {
+          int idx = ( (i + 1) * (grid_size + 2) + (j + 1)) * (grid_size + 2) + (k + 1);
+
+          for (int l = 1; l <= hists_size; ++l, ++pos)
+          {
+            output.points[0].histogram[pos] = hists[idx][l];
+          }
```

Comment:
But hists[idx] is an Eigen::VectorXf. Is it possible to copy elements of Eigen::VectorXf to std::vector using std::copy?